test/connector/tcp/mssql/client_request.go

```diff
@@ -0,0 +1,192 @@
+package mssqltest
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/cyberark/secretless-broker/internal/log"
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp"
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp/mssql"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+type clientRequest struct {
+	params dbConfigParams
+	query  string
+}
+
+type clientResponse struct {
+	out string
+	err error
+}
+
+func cloneCredentials(original map[string][]byte) map[string][]byte {
+	clone := make(map[string][]byte)
+	for key, value := range original {
+		valueCopy := make([]byte, len(value))
+		copy(valueCopy, value)
+
+		clone[key] = valueCopy
+	}
+	return clone
+}
+
+func (clientReq clientRequest) proxyRequest(
+	executor dbQueryExecutor,
+	credentials map[string][]byte,
+) (string, string, error) {
+
+	// Copy credentials. This prevents any mutation or zeroization
+	credentials = cloneCredentials(credentials)
+
+	var err error
+
+	logger := log.New(true)
+	testLogger := logger.CopyWith("[TEST]", true)
```

Comment:
Maybe [MSSQL TEST]? Might be more wordy, but could help when digging through logs